src/Appwrite/Platform/Workers/Webhooks.php

```diff
@@ -117,10 +133,87 @@ private function execute(array $events, string $payload, Document $webhook, Docu
             \curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
         }
 
-        if (false === \curl_exec($ch)) {
-            $this->errors[] = \curl_error($ch) . ' in events ' . implode(', ', $events) . ' for webhook ' . $webhook->getAttribute('name');
-        }
-
+        $responseBody = \curl_exec($ch);
+        $curlError = \curl_error($ch);
+        $statusCode = curl_getinfo($ch, CURLINFO_RESPONSE_CODE);
         \curl_close($ch);
+
+        if (!empty($curlError) || $statusCode >= 400) {
+            $dbForConsole->increaseDocumentAttribute('webhooks', $webhook->getId(), 'attempts', 1);
+            $webhook = $dbForConsole->getDocument('webhooks', $webhook->getId());
+            $attempts = $webhook->getAttribute('attempts');
+
+            $logs = '';
+            $logs .= 'URL: ' . $webhook->getAttribute('url') . "\n";
+            $logs .= 'Method: ' . 'POST' . "\n";
+
+            if (!empty($curlError)) {
+                $logs .= 'CURL Error: ' . $curlError . "\n";
+                $logs .= 'Events: ' . implode(', ', $events) . "\n";
+            } else {
+                $logs .= 'Status code: ' . $statusCode . "\n";
+                $logs .= 'Body: ' . "\n" . \mb_strcut($responseBody, 0, 10000) . "\n"; // Limit to 10kb
+            }
+
+            $webhook->setAttribute('logs', $logs);
+
+            if ($attempts >= self::MAX_FAILED_ATTEMPTS) {
+                $webhook->setAttribute('enabled', false);
+
+                $memberships = $dbForConsole->find('memberships', [
+                    Query::equal('teamInternalId', [$project->getAttribute('teamInternalId')]),
+                    Query::limit(APP_LIMIT_SUBQUERY)
+                ]);
+
+                $userIds = array_column(\array_map(fn ($membership) => $membership->getArrayCopy(), $memberships), 'userId');
+
+                $users = $dbForConsole->find('users', [
+                    Query::equal('$id', $userIds),
+                ]);
+
+                $protocol = App::getEnv('_APP_OPTIONS_FORCE_HTTPS') == 'disabled' ? 'http' : 'https';
+                $hostname = App::getEnv('_APP_DOMAIN');
+                $projectId = $project->getId();
+                $webhookId = $webhook->getId();
+
+                $template = Template::fromFile(__DIR__ . '/../../../../app/config/locale/templates/email-webhook-failed.tpl');
+
+                $template->setParam('{{webhook}}', $webhook->getAttribute('name'));
+                $template->setParam('{{project}}', $project->getAttribute('name'));
+                $template->setParam('{{url}}', $webhook->getAttribute('url'));
+                $template->setParam('{{error}}', $curlError ??  'The server returned ' . $statusCode . ' status code');
+                $template->setParam('{{redirect}}', $protocol . '://' . $hostname . "/console/project-$projectId/settings/webhooks/$webhookId");
+                $template->setParam('{{attempts}}', $attempts);
+
+                $subject = 'Webhook deliveries have been paused';
```

Comment:
shouldn't this be part of the translation files?

Reply:
We are not using a translation file for this email since it's a complex one and cloud only supports english for now. So all the `body` of the email is in `.tpl` file and `subject` is in this file.